src/daemon/rpmostreed-transaction-types.cxx

```diff
@@ -2752,10 +2755,36 @@ kernel_arg_transaction_execute (RpmostreedTransaction *transaction,
       ostree_kernel_args_append_argv (kargs, self->kernel_args_added);
     }
 
+
   /* After all the arguments are processed earlier, we convert it to a string list*/
   g_auto(GStrv) kargs_strv = ostree_kernel_args_to_strv (kargs);
   rpmostree_sysroot_upgrader_set_kargs (upgrader, kargs_strv);
 
+  if (append_if_missing)
+    {
+      for (char **iter = &append_if_missing; iter && *iter; iter++)
+        {
+          const char *arg = *iter;
+          if(!g_strv_contains(kargs_strv,arg))
```

Comment:
Hmm OK right, this is a bit tricky. I think ideally we would add more APIs to `OstreeKernelArgs` like `ostree_kernel_args_append_if_missing`. I recognize though that this is increasing the scope of this supposed "good first issue" quite a bit, so for now we can do this instead:

```c
g_auto(GStrv) existing_kargs = g_strsplit (self->existing_kernel_args, " ", -1);
...
for (...)
  {
    const char *arg = *iter;
    if (!g_strv_contains (existing_kargs, arg))
      ostree_kernel_args_append (kargs, arg);
  }
...
// same idea for delete_if_present
...
g_auto(GStrv) kargs_strv = ostree_kernel_args_to_strv (kargs);
rpmostree_sysroot_upgrader_set_kargs (upgrader, kargs_strv);
```